nixos/modules/programs/steam.nix

```diff
@@ -69,6 +75,19 @@ in {
       '';
     };
 
+    extraPackages = lib.mkOption {
```

Comment:
This should probably be a function as well, no?

Reply:
No. For package's argument  `extraPkgs` , we may not have the same `pkgs` as which called the pcakcge itself while overriding. So providing a function to pass `pkgs` is necessary.

For NixOS option 'extraPackages', users should always get the same `pkgs` in their NixOS configuration globally, as the flake input or `<nixpkgs>` .

I have checked the already existing `extraPackages` in NixOS Search - options, most of their types is list of packages, the remain, whose type are function that evaluates to a(n) list of package, has all argument is `pythonPackages` or `haskellPakcages`.

Reply:
Does this not get used for multilib as well?

Reply:
> Does this not get used for multilib as well?

Sorry, I don't understand your question. Could you please describe it in detail.

Reply:
Those functions should be called twice, once for 64-bit packages (with the 64-bit package set), and once for 32-bit packages. Or am I wrong here?

Reply:
Oh, I see, I confused it with `extraLibraries`, sorry. This is fine then.